test.py

```diff
@@ -82,39 +82,47 @@ class MySchema(Schema):
             assert hfile['/y'].attrs['desc'] == desc
             assert hfile['/z'].attrs['desc'] == desc
         else:
-            assert len(hfile['/v'].attrs.keys()) == 0
-            assert len(hfile['/w'].attrs.keys()) == 0
-            assert len(hfile['/x'].attrs.keys()) == 0
-            assert len(hfile['/y'].attrs.keys()) == 0
-            assert len(hfile['/z'].attrs.keys()) == 0
+            # 'type' attribute should be populated
+            assert len(hfile['/v'].attrs.keys()) == 1
+            assert len(hfile['/w'].attrs.keys()) == 1
+            assert len(hfile['/x'].attrs.keys()) == 1
+            assert len(hfile['/y'].attrs.keys()) == 1
+            assert len(hfile['/z'].attrs.keys()) == 1
```

Comment:
If there should be a specific attribute, then you should check for its existence rather than just checking that there's an attribute there.